R/FilteredDataset.R

```diff
@@ -140,29 +117,27 @@ FilteredDataset <- R6::R6Class( # nolint
     #' @description
     #' Initializes this `FilteredDataset` object
     #'
-    #' @param dataset (`TealDataset`)\cr
+    #' @param dataset (`data.frame` or `MultiAssayExperiment`)\cr
     #'  single dataset for which filters are rendered
-    initialize = function(dataset) {
-      checkmate::assert_class(dataset, "TealDataset")
-      logger::log_trace("Instantiating { class(self)[1] }, dataname: { deparse1(dataset$get_dataname()) }")
+    #' @param dataname (`character`)\cr
+    #'  A given name for the dataset it may not contain spaces
+    #' @param keys optional, (`character`)\cr
+    #'   Vector with primary keys
+    #' @param label (`character`)\cr
+    #'   Label to describe the dataset
+    #' @param metadata (named `list` or `NULL`) \cr
+    #'   Field containing metadata about the dataset. Each element of the list
+    #'   should be atomic and length one.
+    initialize = function(dataset, dataname, keys = character(0), label = attr(dataset, "label"), metadata = NULL) {
+      # TODO assertions
+
+      logger::log_trace("Instantiating { class(self)[1] }, dataname: { deparse1(dataname) }")
       private$dataset <- dataset
-
-      dataname <- self$get_dataname()
-      private$reactive_data <- reactive({
-        env <- new.env(parent = parent.env(globalenv()))
-        for (idx in seq_along(private$eval_env)) {
-          env[[names(private$eval_env)[idx]]] <- if (is.reactive(private$eval_env[[idx]])) {
-            private$eval_env[[idx]]()
-          } else {
-            private$eval_env[[idx]]
-          }
-        }
-        env[[dataname]] <- self$get_data(filtered = FALSE)
-        filter_call <- self$get_call()
-        eval_expr_with_msg(filter_call, env)
-        get(x = self$get_filtered_dataname(), envir = env)
-      })
-
+      private$dataname <- dataname
+      private$keys <- keys
+      private$label <- if (is.null(label)) character(0) else label
+      private$metadata <- metadata
+      private$calculate_hash()
```

Comment:
can the hash be an active field recalculated on request? - we would have less fields/slots to maintain

Reply:
I guess it depends on how quick it is to hash a dataset - when it's large could it take too long?

Reply:
@pawelru I think we don't need this field to be dynamic now as unfiltered data is set once "forever". When we finaly figure out how to reload DDL this would be needed. At this point it's not relevant to add complexity to this one.